### PR TITLE
[UNR-5783] Fix test reruns

### PR DESCRIPTION
**Pull request by mironec**
#### Description
Fix test re-running.

#### Release note
N/A

#### Tests
Ran locally.

#### Documentation
N/A

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp [hunk 1]
```diff
@@ -746,6 +747,11 @@ void ASpatialFunctionalTest::PrepareTestAfterBeginPlay()
 	{
 		PrepareTest();
 	}
+
```
Comment (Vix2021) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:750]: Please can we add a comment here to say that this is for re-running tests?
So when we come back to this in the future we understand why this is needed.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestFlowController.cpp [hunk 2]
```diff
@@ -49,20 +49,14 @@ void ASpatialFunctionalTestFlowController::OnActorReady(bool bHasAuthority)
...
 		// Registration of authoritative flow controllers (server and client)
 		OwningTest->RegisterFlowController(this);
-
-		if (WorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server)
-		{
-			TrySetReadyToRunTest();
-		}
 	}
+
+	TrySetReadyToRunTest();
```
Comment (Vix2021) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestFlowController.cpp:54]: This only needs to be called if the flow controller has authority, so should be moved inside the check above.

**Review by Vix2021 — APPROVED** [review]
Thanks for fixing this.
I made a couple of minor suggestions but otherwise looks good :-)

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp (hunk 1 above)
Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:750]: Added a comment

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestFlowController.cpp (hunk 2 above)
Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestFlowController.cpp]: Moved it.
Originally, I wanted to remove the clientOwnership trigger completely, but it didn't pan out.
I still think it should somehow be possible to make this simpler, but I think I would rather just refactor the entire framework.

**Pull request merged**